app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -19,9 +19,11 @@ window.GOVUK.analyticsGA4.analyticsModules = window.GOVUK.analyticsGA4.analytics
         this.handleClick = this.handleClick.bind(this)
         this.handleMousedown = this.handleMousedown.bind(this)
 
-        document.querySelector('body').addEventListener('click', this.handleClick)
-        document.querySelector('body').addEventListener('contextmenu', this.handleClick)
-        document.querySelector('body').addEventListener('mousedown', this.handleMousedown)
+        if (!this.config.disableListeners) {
```

Comment:
Which other modules are using this code?

Reply:
@andysellick `ga4-enhanced-ecommerce-tracker.js` is using this code to check if a link is external.

Theoretically, `ga4-enhanced-ecommerce-tracker.js` might work without having to initialise it's own link tracker instance, since `ga4-link-tracker.js` would initialise on each pageload. However then it would rely on `ga4-link-tracker.js` loading before `ga4-enhanced-commerce-tracker.js` which seems a bit risky. 

Therefore, I was thinking we would let `ga4-enhanced-ecommerce-tracker.js` initialise it's own link tracker, but without any click listeners. What do you think about this?

Reply:
@andysellick you might be right that all links in GOV.UK search are internal, but I'm not sure about the other finders. Since all the finders share the same template, I believe the eCommerce code is going to run on all the finders as soon as it's deployed even though we only scoped in GOV.UK search.

Basically, when you click on a search result, the eCommerce GTM object contains:

```JSON
"event_data": {
    "external": "true/false"
}
```

Which uses `Ga4LinkTracker.isExternalLink()` to determine wether the link is external or not. And in order to determine if something is external, the link tracker module needs to know what is classed as internal, which is grabbed when it is initialised. It relies on multiple other functions inside the link tracker too, which is why one function couldn't be simply duplicated into `ga4-enhanced-ecommerce-tracker`.

I think you might be correct that we won't need to initialise `ga4-link-tracker.js` again, but then the eCommerce code will only work if `ga4-link-tracker.js` has initialised first so it creates a dependency on the run order. This is because when `init()` is called on `ga4-link-tracker.js`, it sets what the internal domains are, the download domains, the assets paths, etc. which are needed to run `.isExternalLink()`. So if this has run before the eCommerce tracking, we wouldn't need to initialise another one. 

However a benefit of having two initialisations is that the search code may have different domains/paths which it wants to determine as internal/download links - and this would work as we can pass through a different config object to this initialisation. I'm not sure if that would ever be needed though.

Reply:
I think I'm happy to go ahead with this as it is but the close linking of these two bits of code feels brittle. Let's include this when we look at refactoring this.

(having thought about it I'm fairly sure all of our search results are internal links, but @gclssvglx or @Rosa-Fox might know for certain)